### PR TITLE
OHRI-1350 Move custom side nav links to utilize openmrs homepage

### PR DESCRIPTION
**Pull request by CynthiaKamau**
## Requirements

- [ ] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes the ticket number in the format `OHRI-123 My PR title`.
- [ ] My work includes tests or is validated by existing tests.

## Summary

Move custom side nav links to utilize openmrs homepage

For each package, the following new files have been created/edited :
`rootComponent.tsx` (new file) :   Initializes the main React application with routing using the react-router-dom library. Additionally, it configures the global settings for data fetching using SWR. The root component sets up a single route ('/') that renders the Home component, providing the basic structure for the application.

`homeComponent.tsx` (new file) : Consolidates various components including the header, metric tiles, and tables that collectively form the specific content of the home page. This component encapsulates the layout and behavior of the home page, making it modular and easier to manage.

`routes.json` (edited file) : The routes.json file has been edited to attach all extensions related to the home page to the homepage-dashboard-slot. This consolidation simplifies the structure by centralizing the home page extensions and makes it more manageable.

`index.ts` (edited file) : In the index.ts file, individual header, metric, and tab/table extensions have been replaced with a single rootComponent. This restructuring enhances code organization and reduces redundancy by consolidating the various extensions into a cohesive and streamlined architecture.

To test this, use this config [here](https://github.com/UCSF-IGHS/openmrs-distro-referenceapplication/tree/OHRI-1350)
```
{
  "@openmrs/esm-home-app": {
    "extensionSlots": {
      "homepage-dashboard-slot": {
        "order": [
          "home-widget-db-link",
          "clinical-appointments-dashboard-link",
          "dispensing-dashboard-link",
          "service-queues-dashboard-link",
          "billing-dashboard-link",
          "ohri-home-programs-text",
          "covid-cases-dashboard-ext",
          "maternal-child-health-results-summary",
          "tb-cases-dashboard-ext",
          "ohri-hiv-prevention-dashboard-ext",
          "care-and-treatment-dashboard-ext"
        ],
        "add": [
          "dispensing-dashboard-link"
        ],
        "remove": [
          "patient-lists-dashboard-link"
        ]
      },
      "homepage-widgets-slot": {
        "remove": [
          "home-appointments"
        ],
        "add": [
          "ohri-patient-list"
        ]
      }
    }
  }
}

```
## Screenshots

https://github.com/UCSF-IGHS/openmrs-esm-ohri/assets/19533785/812e45a8-fdd5-4dec-824d-1dbb9d1ae752

### CODE REVIEW

**Comment by pirupius** [comment]
Hey @CynthiaKamau the UI on the screenshot looks mis-aligned. The Maternal sidenav is meant to be at the same level as the other programme links

**Comment by CynthiaKamau** [comment]
> Hey @CynthiaKamau the UI on the screenshot looks mis-aligned. The Maternal sidenav is meant to be at the same level as the other programme links

Hi, ive fixed that

File: packages/esm-commons-lib/src/index.ts [hunk 1]
```diff
@@ -29,6 +29,8 @@ export * from './components/tile/ohri-summary-tile-tablet.component';
 export * from './components/tile/ohri-summary-tile.component';
 export * from './utils/compare';
 export * from './utils/createOHRIDashboardLink';
+export * from './utils/createNewOHRIDashboardLink';
+export * from './utils/createOHRIGroupedLink';
```
Comment (pirupius) [inline packages/esm-commons-lib/src/index.ts:33]: Can we move these to the openmrs?

File: packages/esm-commons-lib/src/utils/createOHRIGroupedLink.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    const pathArray = location.pathname.split('/home');
+    const lastElement = pathArray[pathArray.length - 1];
+    return decodeURIComponent(lastElement);
+  }, [location.pathname]);
+
+  if (isHidden) {
+    return;
+  }
+
+  if (isFolder) {
```
Comment (pirupius) [inline packages/esm-commons-lib/src/utils/createOHRIGroupedLink.tsx:34]: This is an implementation that other people can take advantage of. Ampath are interested in this as well.

Comment (CynthiaKamau) [reply packages/esm-commons-lib/src/utils/createOHRIGroupedLink.tsx:34]: i can do that in a follow up PR

File: packages/esm-covid-app/src/home.component.tsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+import CovidSummaryTiles from './views/dashboard/summary-tiles/covid-summary-tiles.component';
+
+const Homecomponent = () => {
+  return (
+    <div style={{ margin: '1rem' }}>
+      <OHRIWelcomeSection title="COVID-19 Cases" />
+      <CovidSummaryTiles />
+      <CovidHomePatientTabs />
+    </div>
+  );
```
Comment (pirupius) [inline packages/esm-covid-app/src/home.component.tsx:13]: I believe this where the designs vary

**Review by pirupius — CHANGES_REQUESTED** [review]
Thanks @CynthiaKamau for the great work done so far. There seems to be some missing designs that were ignored during the migration to the app dashboard approach. This is how the dashboard should look like putting into consideration the spacing between theh tabs and tiles, alignment of tabs with the sidenav, etc

How it's meant to be:

Screenshot 2024-03-12 at 09 03 22

How it looks like in this PR:

Screenshot 2024-03-12 at 09 05 20

File: packages/esm-covid-app/src/home.component.tsx (hunk 3 above)
Comment (CynthiaKamau) [reply packages/esm-covid-app/src/home.component.tsx:13]: Let me address that

Comment (CynthiaKamau) [reply packages/esm-covid-app/src/home.component.tsx]: https://github.com/UCSF-IGHS/openmrs-esm-ohri/assets/19533785/b9a03681-bd76-4df7-82e9-c2fa37f30538

This is the updated video, thoughts @pirupius

**Review by pirupius — APPROVED** [review]

**Pull request merged**